source/dofs/dof_handler_policy.cc

```diff
@@ -4300,6 +4925,13 @@ namespace internal
             triangulation->coarse_cell_to_p4est_tree_permutation,
             triangulation->p4est_tree_to_coarse_cell_permutation);
 
+          // in case of hp::DoFHandlers, we may have received valid
+          // indices of degrees of freedom that are dominated by a fe
+          // object on a ghost cell.
+          // thus, we overwrite the remaining invalid indices on our
+          // locally owned cell with the valid ones in this step.
+          Implementation::merge_invalid_dof_indices_on_ghost(*dof_handler);
```

Comment:
given the comment, the function is not correctly named because you are not actually doing the merge *on* the ghost, but you are merging *from* the ghost. is this correct? A long but correct name would be `merge_invalid_locally_active_dofs_from_ghost_cells()`. I'm not saying that this should be the name, but would this be correct?

Reply:
The line below this one is what in the distributed paper is called "Phase 5" if I recall correctly, but it does not have a comment to this effect. Would you mind adding that as well?

Reply:
It depends on which cell is dominating across the ghost interface. The dominating cell distributes its indices to the dominated one. I would suggest to clarify it by calling it `merge_invalid_dof_indices_on_ghost_interfaces()`.

I have a nice example in mind that we can use in the upcoming paper. This will clarify things :-)

Reply:
Great. Choose a name you think is good.

Example: Fantastic! We need these kinds of examples to illustrate things.